core/src/main/java/com/infinum/jsonapix/core/discriminators/BaseJsonApiDiscriminator.kt

```diff
@@ -92,10 +93,12 @@ abstract class BaseJsonApiDiscriminator(
         val relationshipsLinksDiscriminator = CommonDiscriminator(relationshipsLinks)
         original.jsonObject.entries.filter { it.value is JsonObject }.forEach { relationshipEntry ->
             val set = relationshipEntry.value.jsonObject.entries.toMutableSet()
-            getLinksObject(relationshipEntry.value)?.let { linksSafe ->
-                val newLinks = relationshipsLinksDiscriminator.inject(linksSafe)
-                set.removeAll { it.key == JsonApiConstants.Keys.LINKS }
-                set.add(getJsonObjectEntry(JsonApiConstants.Keys.LINKS, newLinks))
+            getLinksObject(relationshipEntry.value)?.let { linksObject ->
+                if (linksObject !is JsonNull) {
+                    val newLinks = relationshipsLinksDiscriminator.inject(linksObject)
+                    set.removeAll { it.key == JsonApiConstants.Keys.LINKS }
+                    set.add(getJsonObjectEntry(JsonApiConstants.Keys.LINKS, newLinks))
```

Comment:
Not sure about this, but do we need to do the same for meta?

Reply:
Yes we do, missed that, will add the handling. Good catch 👍